File: 0007. Reverse Integer/solution.py
# Brute Force Solution
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        string = ''
        if x < 0:
            x *= - 1
            string += '-'
        elif x == 0:
            string += '0'

        while x > 0:
            string += str(x % 10)
            x = x // 10

        if -2**31 <= int(string) <= (2 ** 31) - 1:
            return int(string)
        else:
            return 0
        
# Optimized solution
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x < 0 : mult = -1
        else : mult = 1

        temp = abs(x)
        res = 0

        while temp > 0:
            res = (res * 10) + temp % 10
            print(res)
            temp //= 10

        res *= mult

        if not (-1 * (2 ** 31) <= res <= (2 ** 31) - 1) : return 0

        return res

File: 0007. Reverse Integer/solution.cpp
#define INT_MIN -2147483648
#define INT_MAX 2147483647

// Optimized Solution
class Solution {
public:
    int reverse(int x) {
        int mult;

        if (x < 1){
            mult = -1;
        }
        else{
            mult = 1;
        }

        x = mult * (long) x;

        long long res = 0;

        while (x > 0){
            res = res * 10 + x % 10;
            x /= 10;
        }

        if ((res > INT_MAX) || (res < INT_MIN)) return 0;
        
        return res * mult;
        
    }
};